.github/workflows/reusable_build.yml

```diff
@@ -0,0 +1,75 @@
+### For the pure soul wishes to move it to another place
+# https://github.com/orgs/community/discussions/9050
+
+name: Build ClickHouse
+'on':
+  workflow_call:
+    inputs:
+      build_name:
+        description: the value of build type from tests/ci/ci_config.py
+        required: true
+        type: string
+      checkout_depth:
+        description: the value of the git shallow checkout
+        required: false
+        type: number
+        default: 1
+      runner_type:
+        description: the label of runner to use
+        default: builder
+        type: string
+      additional_envs:
+        description: additional ENV variables to setup the job
+        type: string
+
+jobs:
+  Build:
+    name: Build-${{inputs.build_name}}
+    runs-on: [self-hosted, '${{inputs.runner_type}}']
+    steps:
+      - name: Check out repository code
+        uses: ClickHouse/checkout@v1
+        with:
+          clear-repository: true
+          submodules: true
+          fetch-depth: ${{inputs.checkout_depth}}
+          filter: tree:0
+      - name: Set build envs
+        run: |
+          cat >> "$GITHUB_ENV" << 'EOF'
+          IMAGES_PATH=${{runner.temp}}/images_path
+          GITHUB_JOB_OVERRIDDEN=Build-${{inputs.build_name}}
+          ${{inputs.additional_envs}}
+          EOF
+          python3 "$GITHUB_WORKSPACE"/tests/ci/ci_config.py --build-name "${{inputs.build_name}}" >> "$GITHUB_ENV"
+      - name: Common setup
+        uses: ./.github/actions/common_setup
```

Comment:
this common_setup used in one place only, so maybe it's not that common and we'd rather add its content into "Set env" to ease file reading

Reply:
```> git grep 'REPO_COPY' .github/workflows/  | wc
    523    1396   45155```

It's spread there very widely

Reply:
And, as well, there's a fuser for the `GITHUB_JOB_OVERRIDDEN`. Unfortunately, it still fails to see if the value is wrong.. Sad story of GH actions

Reply:
After some research, I've added there as well the repository copy. Less code for the God of DRY!

Reply:
I don't see an other places where common_setup is called. I suggest grep for common_setup, not REPO_COPY

Reply:
I won't touch the old stuff to add it in 500 places. But it will be used in every `reusable_*.yml`